### ISSUE TITLE
Strange looking numbers

### ISSUE THREAD

**Issue by beli3ver** [Kozea/WeasyPrint#758]
First of all, HAPPY NEW YEAR :tada: 

I have a little problem, and I hope you can help me.
I use weasyprint to generate from HTML to PDF my invoice, but the numbers look really strange:

![weasy](https://user-images.githubusercontent.com/7816755/50571053-f6051480-0d9f-11e9-8573-ca3543b854c4.png)

I changed my serif font:

```bash
malte@pinetree:~/Dokumente/Buchhaltung$ fc-match serif
LiberationSerif-Regular.ttf: "Liberation Serif" "Regular"
```
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<!-- -->
<fontconfig>
  <!-- Generic name aliasing -->
  <alias>
    <family>sans-serif</family>
    <prefer>
      <family>Liberation Sans</family>
    </prefer>
  </alias>
  <alias>
    <family>serif</family>
    <prefer>
      <family>Liberation Serif</family>
    </prefer>
  </alias>
  <alias>
    <family>monospace</family>
    <prefer>
      <family>Liberation Mono</family>
    </prefer>
  </alias>
</fontconfig>
```
But still the same strange looking numbers.
Here some system infos:

```bash
malte@pinetree:~/Dokumente/Buchhaltung$ python2 --version
Python 2.7.15+
malte@pinetree:~/Dokumente/Buchhaltung$ weasyprint --version
WeasyPrint version 0.42.3
```
My system is **Debian unstable**
My generation script is still python2

**Comment by liZe** [comment]
Hello @beli3ver, happy new year too!

What a strange bug. Looks like your numbers use a bitmap font.

I think that the numbers in your HTML file are not "normal" numbers but [fullwidth](https://en.wikipedia.org/wiki/Halfwidth_and_fullwidth_forms) numbers. These numbers are sometimes used (for bad reasons) in invoices because figures have the same (huge) width and are thus aligned. Liberation probably doesn't include these numbers, that's why you get a fallback using another (bitmap) font.

If you really want to align numbers, you can use [tabular numbers](https://www.w3.org/TR/css-fonts-3/#font-variant-numeric-prop) if your font provides this feature.

(And if you don't understand at all what I'm talking about, no problem, just post your HTML file in a comment, I'll check that for you :wink:.)

**Comment by beli3ver** [comment]
I think I choose the second option. I can send the python part two, when you need it. Thanks
```html
<html>
    <head>
        <link href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-Zug+QiDoJOrZ5t4lssLdxGhVrurbmBWopoEl+M6BdEfwnCJZtKxi1KgxUyJq13dy" crossorigin="anonymous">
          <style>
@page {
    margin: 3cm 2cm 3cm 2cm;

    @top-right {
        content: url("logo.png");
    }

    @bottom-left {
        white-space: pre;
        text-align: left;
        font-size: 8pt;
        content: "KIEFER NETWORKS" "\a"
                 "sdf" "\a"
                 "sdf" "\a"
                 "sdf" "\a"
                 "USt-IdNr: sdf" "\a\a\a"
    }

    @bottom-center {
        white-space: pre;
        text-align: left;
        font-size: 8pt;
        content: "www.Kiefer-Networks.de" "\a"
                 "sdf" "\a"
                 "sdf"
                 "\a\a"
                 "Seite " counter(page)
                 " von " counter(pages)
    }

    @bottom-right {
        white-space: pre;
        text-align: left;
        font-size: 8pt;
        content: "Isdf" "\a"
                 "sdf" "\a"
                 "sdf" "\a"
                 "\a""\a"

    }
}

html {
    font-size: 10pt;
    font-family: serif;
}

.section {
    margin-top: 1cm;
}

/**
 * Address table
 */
 .adressTable tr:nth-child(1) td {
     border-bottom: 1px solid black;
 }

 .small {
     font-size: 8pt;
 }

/**
 * Info table
 */
.infoTable {
    margin-top: 0.5cm;
    width: 35%;
    float: right;
    margin-bottom: 2cm;
    text-align: right;
}

.infoTable thead {
    font-weight: bold;
    text-align: left;
    font-size: 18px;
}

.infoTable thead td {
      border-bottom: 1px solid black;
}

/**
 * Invoice table
 */
 .invoiceTable tr:nth-child(even) {
     background-color: #efefef;
 }
.invoiceTable {
    width: 100%;
}

.invoiceTable td {
    padding: 5px;
    width: 85%;
}

.invoiceTable thead {
    font-weight: bold;
    border-bottom: 1px solid black;
}

.invoiceTable .padrow {
    height: 35px;
}

.invoiceTable .tfoot td {
    border: none;
    padding: 2px;
}


.invoiceTable2 {
    width: 100%;
}

.invoiceTable2 td {
    padding: 5px;
    width: 85%;
}

.invoiceTable2 thead {
    font-weight: bold;
    border-bottom: 1px solid black;
}

.invoiceTable2 .padrow {
    height: 35px;
}

.invoiceTable2 .tfoot td {
    border: none;
    padding: 2px;
}

.nowrap {
    white-space: nowrap;
    width: 5%;
}

.smallNum {
    font-weight: normal;
    font-size: 10px;
    text-align: right;
}

.bold {
    font-weight: bold;
    font-size: 14px;
    text-align: right;
}
.txtleft {
  text-align: left!important;
}

.txtright {
  text-align: right!important;
}

.txtcenter {
  text-align: center!important;
}

          </style>
          </head>
    <body>
        <div class="section">
            <table class="adressTable">
                <tr>
                  <td class="small">{{from.name}}, {{from.street}}, {{from.postcode}} {{from.city}}</td>
                </tr>
                <tr><td><strong>{{to.name}}</strong></td></tr>
                <tr><td>{{to.contact}}</td></tr>
                <tr><td>{{to.street}}</td></tr>
                <tr><td>{{to.postcode}} {{to.city}}</td></tr>
            </table>
        </div>

        <div class="section">
            <table class="infoTable">
              <thead>
                  <tr>
                      <td colspan="2">Rechnung</td>
                  </tr>
              </thead>
                <tr>
                    <td class="txtleft">Rechnungsnummer: </td>
                    <td>{{invoice.number}}</td>
                </tr>
                <tr>
                    <td class="txtleft">Kundennummer: </td>
                    <td>{{customer_number}}</td>
                  </tr>
                  <tr>
                    <td class="txtleft">Rechnungsdatum: </td>
                    <td>{{invoice.date}}</td>
                </tr>
                <tr>
                  <td class="txtleft">Fälligkeitsdatum: </td>
                  <td>{{invoice.pay_until_date}}</td>
              </tr>
            </table>
        </div>

        <div class="section">
            <table class="invoiceTable">
                <thead>
                    <tr>
                        <td>Beschreibung</td>
                        <td class="txtcenter">Menge</td>
                        <td class="txtcenter">Einheit</td>
                        <td class="txtcenter">Preis</td>
                        <td class="txtcenter">Betrag</td>
                    </tr>
                </thead>
                <tbody>
                    {{#positions}}
                    <tr>
                        <td style="width: 25%;overflow: hidden;">{{{text}}}</td>
                        <td class="nowrap txtcenter">{{amount}}</td>
                        <td class="nowrap txtcenter">{{amount_desc}}</td>
                        <td class="nowrap txtright">{{netto_price}}</td>
                        <td class="nowrap txtright">{{total_netto_price}}</td>
                    </tr>
                    {{/positions}}
                    <tr class="tfoot">
                        <td class="padrow"></td>
                    </tr>
                </tbody>
              </table>
                <table class="invoiceTable2">
                  <tbody>
                    <tr class="tfoot">
                        <td class="noborder"></td>
                        <td colspan="2"> <small>Nettogesamt</small></td>
                        <td class="noborder"></td>
                        <td class="nowrap smallNum">{{totals.netto}} {{currency}}</td>
                    </tr>
                    <tr class="tfoot">
                        <td class="noborder"></td>
                        <td colspan="2"> <small>Umsatzsteuer 19%</small></td>
                        <td class="noborder"></td>
                        <td class="nowrap smallNum">{{totals.tax}} {{currency}}</td>
                    </tr>
                    <tr class="tfoot">
                        <td class="noborder"></td>
                        <td colspan="2"> Rechnungsbetrag</td>
                        <td class="noborder"></td>
                        <td class="nowrap bold">{{totals.brutto}} {{currency}}</td>
                    </tr>
                </tbody>
            </table>
        </div>

        <div class="section">
            <p>Vielen Dank für Ihren Auftrag</p>
        </div>

    </body>
</html>

```

**Comment by liZe** [comment]
> I think I choose the second option.

:smile:

Unfortunately, I can't help you with the template, I need a generated HTML sample with real numbers (instead of `netto_price`, `total_netto_price`…).

**Comment by beli3ver** [comment]
Ok found the bug :dancer:
I removed this line, no it looks perfect :+1:

```html
<link href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-Zug+QiDoJOrZ5t4lssLdxGhVrurbmBWopoEl+M6BdEfwnCJZtKxi1KgxUyJq13dy" crossorigin="anonymous">
```

**Comment by liZe** [comment]
For future readers: this problem is also fixed in Pango 1.57.0.
